src/app/utils/useWhatChanged.ts

```diff
@@ -0,0 +1,25 @@
+import { useRef } from 'react';
+
+/**
+ * A debugging hook that will print out what changed since the last render!
+ *
+ * Example:
+ *
+ * useWhatChanged('MyComponent', {prop1, prop2, state1, state2});
+ */
+export function useWhatChanged<T extends object>(name: string, params: T) {
```

Comment:
This is really cool, there has been a few times when this would have been really useful.

Reply:
Yeah I found it immediately useful! Way better than random console logs